### PR TITLE
Updated Documentation for DevExchange Vault Service and implementation.

### PR DESCRIPTION
**Pull request by bashbang**
My team just completed an implementation of the DevExchange Vault service offering.  We've complied some notes and examples of how we did the implementation.

### CODE REVIEW

File: vault/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+The DevExchange group provisions a Hashicorp Vault for each Openshift (OC4) Project Set (licensepate-dev/test/prod/tools - eg: abc123-dev, abc123-test, abc123-prod, abc123-tools).
+
+To start, you need to have permissions to use the Vault. What we've been able to gather is that whomever is the "owner" of the Openshift environment is generally also the "technical contact". This same person (or persons) are also owner(s) of the Vault. This is typically the person who requested the OC4 Project Set.  Within the Vault service the only people who are able to grant additional permissions is the DevExchange team. The owner does not have rights to alter roles. Best method to get additional technical resources (eg: Devops Engineers) is to ask the owner to post a request in the [#devops-vault RocketChat channel](https://chat.developer.gov.bc.ca/channel/devops-vault). As of the time of writing, users do not have permissions to create additional secret engines. You must therefore organize your secrets into the pre-allocated secret engines. This constraint may cause issues if your secrets have the same name throughout the different namespaces.
```
Comment (StevenBarre) [inline vault/README.md]: I believe it's the Product Owner and Tech Lead specified in the Project Registryhttps://registry.developer.gov.bc.ca/

File: vault/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+To log into the vault start here:
+[https://vault.developer.gov.bc.ca/](https://vault.developer.gov.bc.ca/)
+
+To log in use the following values:
+Namespace: platform-services
+Method: OIDC
+Role: {LicensePlate}  (eg: abc123) (Note: No need for the dev/test/prod suffix)
```
Comment (StevenBarre) [inline vault/README.md]:
```suggestion
To log into the vault start here: [https://vault.developer.gov.bc.ca/](https://vault.developer.gov.bc.ca/)

To log in use the following values:

- Namespace: platform-services
- Method: OIDC
- Role: {LicensePlate}  (eg: abc123) (Note: No need for the dev/test/prod suffix)
```

Improve Markdown rendering

File: vault/README.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+-     |---prod_service_account          #secret data
+-     |---prod_service_account_pass     #secret data
+```
+
+Note: Don't use a hyphen in a "key".  Vault will accept the value but it's problematic in the Openshift templates which shows up as an error in the vault-init container.
+
+# Usage of the Vault secrets in Openshift.
+When you look at the getting-started-demo you'll see there's already a deployment template to assist. The two important parts of this vault demo are:
+    annotations
+    serviceaccount
```
Comment (StevenBarre) [inline vault/README.md]:
```suggestion
# Usage of the Vault secrets in Openshift.

When you look at the getting-started-demo you'll see there's already a deployment template to assist. The two important parts of this vault demo are:

- `annotations`
- `serviceaccount`
```
Improve Markdown rendering

File: vault/README.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+## Annotation
+We elected to use the annotation / sidecar method. [This was helpful from Hashicorp](https://www.vaultproject.io/docs/platform/k8s/injector/examples).
+
+Note: Learn from my mistake in the Deployment. Place the annotation in the correct location. In the "Deployment" it belongs in the "metadata" portion of "template" NOT the root metadata section:
+
+```
```
Comment (StevenBarre) [inline vault/README.md]:
```suggestion
```yaml
```
Improve Markdown rendering

File: vault/README.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    spec:
+      serviceAccount: abc123-vault
+      # or for HELM with var replacements use:
+      #serviceAccount: {{ .Values.global.licenseplate }}-vault
+
+  ```
+
+Note the additional line under spec that I added. This is the service account that is needed to connect to the Vault. This account has been created already for you in Openshift so on the surface it's straight forward. You may notice another one that's used in the demo:
+```        serviceAccountName: LICENSE-vault```
+I believe the serviceAccountName is used for Openshift and ServiceAccount is used by Kubernetes. I've yet to determine the differences or why there is a difference, but in any case, using ServiceAccount did the trick for me while using Deployments. Having both did not hurt. This is more the Kubernetes way of doing things than the Openshift method which I've seen uses DeploymentConfigs and possibly that's where serviceAccountName comes into play?
```
Comment (StevenBarre) [inline vault/README.md]:
```suggestion
Note the additional line under spec that I added. This is the service account that is needed to connect to the Vault. This account has been created already for you in Openshift so on the surface it's straight forward. You may notice another one that's used in the demo: `serviceAccountName: LICENSE-vault`
```

According to `oc explain pod.spec.serviceAccount` that one is deprecated and `serviceAccountName` is the new/correct one to use.

Also improved the markdown a bit.

File: vault/README.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        vault.hashicorp.com/agent-inject-secret-microservices-secret-debug: {{ .Values.global.licenseplate }}-{{ .Values.global.vault_engine }}/microservices-secret-debug
+        vault.hashicorp.com/agent-inject-template-microservices-secret-debug: |
+          {{`{{- with secret `}}"{{ .Values.global.licenseplate }}-{{ .Values.global.vault_engine }}/microservices-secret-debug"{{` }}
+          export dev_hostname="{{ .Data.data.dev_hostname }}"
+          export dev_toolbox="{{ .Data.data.dev_toolbox }}"
+          {{- end `}} }}
+    spec:
+      serviceAccount: abc123-vault
+      # or for HELM with var replacements use:
+      #serviceAccount: {{ .Values.global.licenseplate }}-vault
```
Comment (StevenBarre) [inline vault/README.md]:
```suggestion
      serviceAccountName: abc123-vault
      # or for HELM with var replacements use:
      #serviceAccountName: {{ .Values.global.licenseplate }}-vault
```

**Comment by bashbang** [comment]
@ShellyXueHan you approved this in rocket-chat, can you please approve here so I can get this PR merged and closed?  ty!

**Review by ShellyXueHan — APPROVED** [review]

**Comment by bashbang** [comment]
@StevenBarre I don't have write permissions for merging.  Assuming this is still the best place for this content, could you please merge?
ty!

**Comment by StevenBarre** [comment]
Thanks for the contribution!

**Pull request merged**